Geometry/HcalCommonData/interface/HcalDDDRecConstants.h

```diff
@@ -0,0 +1,109 @@
+#ifndef Geometry_HcalTowerAlgo_HcalDDDRecConstants_h
+#define Geometry_HcalTowerAlgo_HcalDDDRecConstants_h
+
+/** \class HcalDDDRecConstants
+ *
+ * this class reads the constant section of
+ * the hcal-sim-numbering xml-file
+ *  
+ *  $Date: 2013/12/25 00:06:50 $
+ * \author Sunanda Banerjee, SINP <sunanda.banerjee@cern.ch>
+ *
+ */
+
+#include<string>
+#include<vector>
+#include<iostream>
+
+#include "CondFormats/GeometryObjects/interface/HcalParameters.h"
+#include "Geometry/HcalCommonData/interface/HcalCellType.h"
+#include "Geometry/HcalCommonData/interface/HcalDDDSimConstants.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
+
+class HcalDDDRecConstants {
+
+public:
+
+  HcalDDDRecConstants(const HcalParameters* hp, const HcalDDDSimConstants& hc);
+  ~HcalDDDRecConstants();
+
+  struct HcalID {
+    int    subdet, eta, phi, depth;
+    HcalID(int sub=0, int et=0, int fi=0, int d=0) : subdet(sub), eta(et),
+						     phi(fi), depth(d) {}
+  };
+  struct HcalEtaBin {
+    int    ieta, nPhi, depthStart;
+    double etaMin, etaMax, phi0, dphi;
+    std::vector<std::pair<int, int> > layer;
+    HcalEtaBin(int eta=0, double et1=0, double et2=0, int nf=0, double fi0=0,
+	       double df=0) : ieta(eta), nPhi(nf),depthStart(0), etaMin(et1), 
+			      etaMax(et2), phi0(fi0), dphi(df) {}
+  };
+  struct HcalActiveLength {
+    int    ieta, depth;
+    double eta, thick;
+    HcalActiveLength(int ie=0, int d=0, double et=0, 
+		     double t=0) : ieta(ie), depth(d), eta(et), thick(t) {}
+  };
+
+  std::vector<std::pair<double,double> > getConstHBHE(const int type) const {
+    if      (type == 0) return gconsHB;
+    else if (type == 1) return gconsHE;
+    else {std::vector<std::pair<double,double> > gcons; return gcons;}
+  }
+  std::vector<int>          getDepth(const int i) const {return layerGroup[i];}
+  std::vector<HcalEtaBin>   getEtaBins(const int itype) const;
+  std::pair<double,double>  getEtaPhi(int subdet, int ieta, int iphi) const;
+  std::pair<int,int>        getEtaRange(const int i) const
+    {return std::pair<int,int>(iEtaMin[i],iEtaMax[i]);}
+  std::vector<double>       getEtaTable()   const {return etaTable;}
+  std::vector<double>       getEtaTableHF() const {return etaTableHF;}
+  std::pair<double,double>  getEtaLimit(const int i) const 
+    {return std::pair<double,double>(etaTable[i],etaTable[i+1]);}
+  HcalID                    getHCID(int subdet, int ieta, int iphi, int lay,
+				    int idepth) const;
+  int                       getMaxDepth(const int type) const {return maxDepth[type];}
+  int                       getNEta() const {return nEta;}
+  double                    getPhiBin(const int i) const {return phibin[i];}
+  double                    getPhiOff(const int i) const {return hpar->phioff[i];}
+  std::vector<double>       getPhiOffs()    const {return hpar->phioff;}
```

Comment:
@bsunanda - it is better to return it by reference. The comment is valid for all member functions in this class and also in HcalDDDSimConstants.